org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/commands/ProjectCommand.java

```diff
@@ -744,8 +747,18 @@ public static void updateProjectSettings(String projectUri, Map<String, Object>
 					}
 					break;
 				default:
+					String settingValue = currentOptions.get(entry.getKey());
+					// only update the options whose keys are valid and value are different from the current ones
+					if (settingValue != null && !settingValue.equals(entry.getValue().toString().trim())) {
```

Comment:
Use `Objects.equals(...)` to compare if it's the same?

Reply:
I want to make sure `settingValue` is not null - if `settingValue` is null, means the option key does not exist in the option map, thus it is not a valid key, then we should skip it.

`Objects.equals(...)` does not guarantee that.